metrics/api/src/main/java/io/helidon/metrics/api/MetricsProgrammaticSettings.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+import java.util.ServiceLoader;
+import java.util.Set;
+
+import io.helidon.common.HelidonServiceLoader;
+import io.helidon.common.LazyValue;
+
+/**
+ * Programmatic (rather than configured) settings that govern certain metrics behavior.
+ * <p>
+ *     Implementations of this interface are typically provided by Helidon itself rather than
+ *     developers building applications and are not intended for per-deployment (or even per-application)
+ *     customization.
+ * </p>
+ */
+public interface MetricsProgrammaticSettings {
```

Comment:
Please add a follow up for naming here - we use `Config` as the suffix for configuring components in other modules, I think it should be used for this class as well